### DISCUSSION TITLE
Cross File References

### DISCUSSION THREAD

**Discussion by Danfs64** [eclipse-langium/langium#861] [Q&A] [answered]
I'm having some issues with Cross-Referencing across two files. I have a language that has A and B nodes on the root, C nodes inside the A nodes and references to the C nodes inside the B nodes (sorry for not providing an actual example, I can provide one if necessary). I had to customize the `ScopeComputation` with a new `computeExports` so I could make these references on the same file, and now I'm testing how to do Cross referencing between files.

The changelog from version 0.0.2 led me to believe that this is done by default, and it certainly looks like it. When I write the second file, the autocomplete for nodes on the first file works, and no "Could not resolve reference" errors appears. But when I try to generate the code from the file, I am greeted with a "Could not resolve reference" on the command-line.

Any directions?

**Comment by msujew** [comment, marked as answer]
Hey @Danfs64,

it looks like your CLI doesn't load your workspace, but only a single file. I believe the CLI that's auto-generated by the yeoman generator only ever works on a single file that it accepts via the `-f` argument. You will need to modify the CLI to load the whole workspace (or relevant files, like with ECMAScript-style imports). We do something similar in our own `langium-cli`, where we only have a single entry point and load all documents that are referenced in one document:

https://github.com/langium/langium/blob/cd5dc74ea1790bf7075e8c5d4f3607a6652b095e/packages/langium-cli/src/generate.ts#L63-L80

**Reply by spoenemann** [reply]
The domainmodel example CLI has a parameter for choosing the workspace root:

https://github.com/langium/langium/blob/cd5dc74ea1790bf7075e8c5d4f3607a6652b095e/examples/domainmodel/src/cli/cli-util.ts#L46-L58

**Reply by Danfs64** [reply]
The domainmodel example did the trick nicely.
Thanks for the help both of you guys :)
